src/nvim/channel.c

```diff
@@ -459,6 +469,7 @@ uint64_t channel_connect(bool tcp, const char *address, bool rpc, CallbackReader
 
   channel->stream.socket.internal_close_cb = close_cb;
   channel->stream.socket.internal_data = channel;
+  channel->stream.socket.on_close = on_close;
```

Comment:
Any reason not to store it on the channel (high level) instead of on the stream (low level)?


```suggestion
  channel->on_close = on_close;
```